samples/navigation-fragments/src/main/java/com/instacart/formula/navigation/CounterFragmentFeatureFactory.kt

```diff
@@ -0,0 +1,34 @@
+package com.instacart.formula.navigation
+
+import com.instacart.formula.android.Feature
+import com.instacart.formula.android.FeatureFactory
+import com.instacart.formula.runAsStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+
+class CounterFragmentFeatureFactory : FeatureFactory<CounterFragmentFeatureFactory.Dependencies, CounterFragmentKey>() {
+
+    interface Dependencies {
+        fun navigationStack(): List<Int>
+        fun counterIncrements(): SharedFlow<Int>
+        fun onNavigateToNext(): () -> Unit
+        fun onNavigateBack(): () -> Unit
+        fun onIncrementCounter(): (Int) -> Unit
```

Comment:
I would probably break this down into `Router` interface.

Reply:
Only 2 properties here are routing-related:
- `onNavigateToNext()`
- `onNavigateBack()`

Do you think it's worth extracting them into a separate interface?